pkg/k8shandler/collection.go

```diff
@@ -180,17 +196,31 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectionPriorityCla
 	return nil
 }
 
-func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectorServiceAccount() error {
+func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectorServiceAccount() (*core.ServiceAccount, error) {
 
 	cluster := clusterRequest.cluster
 
 	collectorServiceAccount := NewServiceAccount("logcollector", cluster.Namespace)
 
 	utils.AddOwnerRefToObject(collectorServiceAccount, utils.AsOwner(clusterRequest.cluster))
 
-	err := clusterRequest.Create(collectorServiceAccount)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Log Collector service account: %v", err)
+	delfinalizer := false
+	if collectorServiceAccount.ObjectMeta.DeletionTimestamp.IsZero() {
+		// This object is not being deleted.
+		if !utils.ContainsString(collectorServiceAccount.ObjectMeta.Finalizers, metav1.FinalizerDeleteDependents) {
+			collectorServiceAccount.ObjectMeta.Finalizers = append(collectorServiceAccount.ObjectMeta.Finalizers, metav1.FinalizerDeleteDependents)
+		}
+		err := clusterRequest.Create(collectorServiceAccount)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return nil, fmt.Errorf("Failure creating Log Collector service account: %v", err)
+		}
+		if len(collectorServiceAccount.ObjectMeta.UID) != 0 {
+			serviceAccountLogCollectorUID = collectorServiceAccount.ObjectMeta.UID
```

Comment:
The UID in the logcollector service account is set when it is created.  So, I store it to use it in ownerReference later here.  If there is an api to retrieve it dynamically, we don't have to store the UID this way...  But so far I could not find it.

sample logcollector service account
```
apiVersion: v1
imagePullSecrets:
- name: logcollector-dockercfg-rgwsp
kind: ServiceAccount
metadata:
  creationTimestamp: "2019-08-15T22:12:46Z"
  finalizers:
  - foregroundDeletion
  name: logcollector
  namespace: openshift-logging
  ownerReferences:
  - apiVersion: logging.openshift.io/v1
    controller: true
    kind: ClusterLogging
    name: instance
    uid: cec9ff31-bfa9-11e9-b67d-029422b5c914
  resourceVersion: "468364"
  selfLink: /api/v1/namespaces/openshift-logging/serviceaccounts/logcollector
  uid: cfa30bf4-bfa9-11e9-a7b2-0a1781cc6f5c
```